instrumentation_test.go

```diff
@@ -127,6 +129,18 @@ func TestOptionPrecedence(t *testing.T) {
 	})
 }
 
+func TestWithResourceAttributes(t *testing.T) {
+	testAttributes := []attribute.KeyValue{
+		semconv.K8SContainerName("test_container_name"),
+		semconv.K8SPodName("test_pod_name"),
+	}
+
+	// Use WithResourceAttributes to config the additional resource attributes
+	c, err := newInstConfig(context.Background(), []InstrumentationOption{WithResourceAttributes(testAttributes...)})
+	require.NoError(t, err)
+	assert.Equal(t, testAttributes, c.additionalResAttrs)
+}
```

Comment:
How about something like

```suggestion
func TestWithResourceAttributes(t *testing.T) {
	attr1 := semconv.K8SContainerName("test_container_name"),
	attr2 := semconv.K8SPodName("test_pod_name"),
	attr3 := semconv.K8SNamespaceName("test_namespace_name"),

	// Use WithResourceAttributes to config the additional resource attributes
	c, err := newInstConfig(context.Background(), []InstrumentationOption{WithResourceAttributes(attr1, attr2), WithResourceAttributes(attr3)})
	require.NoError(t, err)
	assert.Equal(t, []attribute.KeyValue{attr1, attr2, attr3}, c.additionalResAttrs)
}
```

It would avoid a regression bug if `WithResourceAttributes` would replace instead of append.